### DISCUSSION TITLE
RESTv3: How to GET merged pull requests?

### DISCUSSION THREAD

**Discussion by AniLeo** [community/community#24879] [Apps, API and Webhooks]
<p>Hello,</p>
<p><a href="https://developer.github.com/v3/pulls/#list-pull-requests" rel="nofollow noopener">https://developer.github.com/v3/pulls/#list-pull-requests</a> mentions that state can be <strong>open</strong> , <strong>close</strong> d or <strong>all</strong> , however, there is no <strong>merged</strong> state.</p>
<p>How can I list merged pull requests only?</p>
<p>Best Regards,</p>
<p>Ani</p>

**Comment by lifeofguenter** [comment]
<p>searching for this as well :slight_smile:</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>The search API can search “merged pull requests” with “is:merged” operator.<br>
 <a href="https://help.github.com/articles/searching-issues/#search-based-on-the-state-of-an-issue-or-pull-request" rel="nofollow noopener">https://help.github.com/articles/searching-issues/#search-based-on-the-state-of-an-issue-or-pull-request</a></p>
<p>I hope this helps!</p>

**Comment by lifeofguenter** [comment]
<p>would it be easily possible to get the merged status of a branch?</p>

**Comment by AniLeo** [comment]
<p>Hello,</p>
<p>Yes, this is helpful, thanks.</p>
<p>I’ve been looking into it and am almost done with what I want to do. Unfortunately, I can’t find any information on how to sort by merged-asc / merged-desc via the Search API. This is possible in the web version of GitHub.</p>
<p>I’ve read the documentation at <a href="https://help.github.com/articles/sorting-search-results/" rel="nofollow noopener">https://help.github.com/articles/sorting-search-results/</a> and <a href="https://help.github.com/articles/sorting-issues-and-pull-requests/" rel="nofollow noopener">https://help.github.com/articles/sorting-issues-and-pull-requests/</a> but it has no documentation on the aforementioned sorting on the Search API</p>
<p>The query I currently have is</p>
<p><strong>/search/issues?q=repo:rpcs3/rpcs3+is:pr+is:merged+merged:%3E2018-10-01&amp;sort=merged</strong></p>
<p>url: <a href="https://api.github.com/search/issues?q=repo:rpcs3/rpcs3+is:pr+is:merged+merged:%3E2018-10-01&amp;sort=merged" rel="nofollow noopener">https://api.github.com/search/issues?q=repo:rpcs3/rpcs3+is:pr+is:merged+merged:%3E2018-10-01&amp;sort=merged</a></p>
<p><strong>/search/issues?q=repo:rpcs3/rpcs3+is:pr+is:merged+merged:%3E2018-10-01</strong></p>
<p>url: <a href="https://api.github.com/search/issues?q=repo:rpcs3/rpcs3+is:pr+is:merged+merged:%3E2018-10-01" rel="nofollow noopener">https://api.github.com/search/issues?q=repo:rpcs3/rpcs3+is:pr+is:merged+merged:%3E2018-10-01</a></p>
<p><em>Edit1: &amp;sort=merged doesn’t do anything, I wrongly assumed so.</em></p>
<p>And I want to sort by the opposite order, just like it’s possible to do in the web version, as in</p>
<p><strong>is:pr is:merged sort:merged-asc merged:&gt;2018-10-01 </strong></p>
<p>url: <a href="https://github.com/RPCS3/rpcs3/pulls?page=1&amp;q=is%3Apr+is%3Amerged+sort%3Amerged-asc+merged%3A%3E2018-10-01" target="_blank" rel="nofollow noopener">https://github.com/RPCS3/rpcs3/pulls?page=1&amp;q=is%3Apr+is%3Amerged+sort%3Amerged-asc+merged%3A%3E2018-10-01</a></p>
<p>How can I apply such sorting via the Search API?</p>
<p>Best Regards,</p>
<p>Ani</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/anileo">@anileo</a>,</p>
<p>I’m glad that helped! You can include the sort qualifier <code>+sort:author-date-desc</code> for descending and <code>+sort:author-date-asc</code> for ascending.</p>
<p>ie:</p>
<p><a href="https://api.github.com/search/issues?q=repo:rpcs3/rpcs3+is:pr+is:merged+sort:author-date-asc+merged:%3E2018-10-01&amp;sort=merged" rel="nofollow noopener">https://api.github.com/search/issues?q=repo:rpcs3/rpcs3+is:pr+is:merged+sort:author-date-asc+merged:%3E2018-10-01&amp;sort=merged</a></p>
<p>Best,</p>
<p>Andrea</p>

**Comment by AniLeo** [comment]
<p>Hello,</p>
<p>Unfortunately this tag doesn’t sort by merged order.</p>
<p>In fact, after taking a closer look, I noticed merge / closed sort it’s likely not implemented. sort:merge or sort:closed doesn’t do the intended sort on the web interface and on the search API it’s doing nothing (the default sorting by creation date is applied).</p>
<p>For example, a small sample with (currently) 6 PRs</p>
<p>search/issues?q=repo:rpcs3/rpcs3+is:pr+is:merged+merged:%3E2018-11-04+sort=author-date-asc</p>
<p><a href="https://api.github.com/search/issues?q=repo:rpcs3/rpcs3+is:pr+is:merged+merged:%3E2018-11-04+sort=author-date-asc" rel="nofollow noopener">https://api.github.com/search/issues?q=repo:rpcs3/rpcs3+is:pr+is:merged+merged:%3E2018-11-04+sort=author-date-asc</a></p>
<table><tbody><tr>
<td><span>0</span></td>
<td><span>closed_at</span></td>
<td><p><span><span>"2018-11-07T10:06:31Z"</span></span></p></td>
</tr></tbody></table>
<table><tbody><tr>
<td><span>1</span></td>
<td><span>closed_at</span></td>
<td><span><span>"2018-11-06T23:04:23Z"</span></span></td>
</tr></tbody></table>
<table><tbody><tr>
<td><span>2</span></td>
<td><span>closed_at</span></td>
<td><p><span><span>"2018-11-07T12:46:52Z"</span></span></p></td>
</tr></tbody></table>
<table><tbody><tr>
<td><span>3</span></td>
<td><span>closed_at</span></td>
<td><p><span><span>"2018-11-05T14:25:39Z"</span></span></p></td>
</tr></tbody></table>
<table><tbody><tr>
<td><span>4</span></td>
<td><span>closed_at</span></td>
<td><span><span>"2018-11-05T15:20:59Z"</span></span></td>
</tr></tbody></table>
<table><tbody><tr>
<td><span>5</span></td>
<td><span>closed_at</span></td>
<td><span><span>"2018-11-05T11:55:20Z"</span></span></td>
</tr></tbody></table>
<table><tbody><tr>
<td><span>0</span></td>
<td><span>created_at</span></td>
<td><p><span><span>"2018-11-07T07:44:35Z"</span></span></p></td>
</tr></tbody></table>
<table><tbody><tr>
<td><span>1</span></td>
<td><span>created_at</span></td>
<td><span><span>"2018-11-06T22:38:51Z"</span></span></td>
</tr></tbody></table>
<table><tbody><tr>
<td><span>2</span></td>
<td><span>created_at</span></td>
<td><span><span>"2018-11-05T20:16:42Z"</span></span></td>
</tr></tbody></table>
<table><tbody><tr>
<td><span>3</span></td>
<td><span>created_at</span></td>
<td><span><span>"2018-11-03T15:19:19Z"</span></span></td>
</tr></tbody></table>
<table><tbody><tr>
<td><span>4</span></td>
<td><span>created_at</span></td>
<td><span><span>"2018-11-02T18:38:52Z"</span></span></td>
</tr></tbody></table>
<table><tbody><tr>
<td><span>5</span></td>
<td><span>created_at</span></td>
<td><span><span>"2018-11-01T10:32:03Z"</span></span></td>
</tr></tbody></table>
<p>I already tried sort = merged, merged-asc, merged-desc, closed, closed-asc, closed-desc, none seem to make any effect.</p>
<p>Would it be possible to implement sorting by closed/merged date?</p>
<p>Best Regards,</p>
<p>Ani</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hey <a class="mention" href="/anileo">@anileo</a>, </p>
<p>What do you see when: </p>
<p><a href="https://api.github.com/search/issues?q=repo:rpcs3/rpcs3+is:pr+is:merged+sort:author-date-asc+merged:%3E2018-10-01&amp;sort=merged" rel="nofollow noopener">https://api.github.com/search/issues?q=repo:rpcs3/rpcs3+is:pr+is:merged+sort:author-date-asc+merged:%3E2018-10-01&amp;sort=merged</a></p>
<p>Best,</p>
<p>Andrea</p>

**Comment by AniLeo** [comment]
<p>Hello,</p>
<p>I get the same output as I would get with</p>
<p><strong>search/issues?q=repo:rpcs3/rpcs3+is:pr+is:merged+sort:author-date-asc+merged:%3E2018-10-01</strong></p>
<p>I thought &amp;sort=merged did anything because on the web interface adding it (“sort: merged-asc”) actually returns a different order than the default one even though it’s also not the correct one.</p>
<p>I now realized that merge/closed datetime related sorting options are not implemented, because trying any sort that is not implemented (e.g. “sort: asdfgh”) will return the same output as trying “sort: merged-asc”.</p>
<p>What does work differently is</p>
<p><strong>search/issues?q=repo:rpcs3/rpcs3+is:pr+is:merged+sort:author-date-asc+merged:%3E2018-10-01+sort:updated-asc</strong></p>
<p>which shows the PRs ordered by the last update order.</p>
<p>However, this is not the effect I’m looking for, as a comment in a PR will change the last update date. I need to order by the “closed_at” parameter (close/merge datetime).</p>
<p>Best Regards,</p>
<p>Ani</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hmmm, I see what you are saying, there is a limitation the only sort orders we can use are the ones in the drop down below, I can definitely see how helpful it would be to have that option and I’m passing this feedback as a feature request.</p>
<p>  <img src="https://user-images.githubusercontent.com/20666190/181087166-acf45385-e082-4886-84e4-c1a021ee023e.png" alt="Screen Shot 2018-11-08 at 3.14.45 PM.png" data-base62-sha1="opQOGsZcAz8KsAuHy9Ilt0srihw" width="238" height="244"></p>

**Comment by evilbullet** [comment]
<p>is this also possible to have smth like: <strong>is_linked_to_issue:</strong> or <strong>is_related_to_issue:</strong> , for PRs? That would help a lot!</p>
<blockquote>
<p><a class="mention" href="/andreagriffiths11">@andreagriffiths11</a> wrote:</p>
</blockquote>
<blockquote>
<p>The search API can search “merged pull requests” with “is:merged” operator.<br>
 <a href="https://help.github.com/articles/searching-issues/#search-based-on-the-state-of-an-issue-or-pull-request" rel="nofollow noopener">https://help.github.com/articles/searching-issues/#search-based-on-the-state-of-an-issue-or-pull-request</a></p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>I hope this helps!</p>
</blockquote>
<p>Thanks</p>
